### PR TITLE
Campaigns Enhancements

### PR DESCRIPTION
**Pull request by kuiro5**
## What's in here
- Add methods for Campaigns API
- Move #add_next_page_token to Marketo::API::Base
- Add #to_query to convert hashes to Marketo-consumable query params

### Related PRs

- n/a

### Related Stories

- n/a

### Notes

none

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    marketo_api (0.0.7.pre.alpha)
+    marketo_api (0.1.0.pre.alpha)
```
Comment (Atbhatt) [inline Gemfile.lock]: Any particular reason to jump from 0.0.7 to 0.1.0?

File: lib/marketo_api/api/base.rb [hunk 2]
```diff
@@ -46,6 +46,38 @@ def version_guard(version)
...
+      # Internal: Append a nextPageToken to a url if token is present
+      #
+      # Examples
+      #
+      #   add_next_page_token('activities.json?')
+      #   # => '/rest/v1/leads'
+      def add_next_page_token(path, token)
+        path += "&nextPageToken=#{token}" if token
+        path
+      end
```
Comment (Atbhatt) [inline lib/marketo_api/api/base.rb]: Nice job moving this out to base!

File: lib/marketo_api/api/base.rb [hunk 3]
```diff
@@ -46,6 +46,38 @@ def version_guard(version)
...
 
+      # Internal: Converts a hash into a string that can be consumed by
+      # the Marketo API as a query string
+      #
+      # Examples
+      #
+      #   to_query({ source: 'Sales Engage', workspaceName: 'Global' })
+      #   # => 'source=Sales%20Engage&workspaceName=Global'
+      def to_query(options = {})
+        options.to_query.gsub('+', '%20')
```
Comment (Atbhatt) [inline lib/marketo_api/api/base.rb]: Is it possible to provide feedback to Logan about this? I don't believe this is standard practice for API's and would be better to get aligned on standard practices before the API is heavily used.

Comment (kuiro5) [reply lib/marketo_api/api/base.rb]: Syncing with him and Yifu now, will let you know where we land.

Comment (kuiro5) [reply lib/marketo_api/api/base.rb]: Got a hold of Yifu, his thoughts are similarly aligned with this StackOverflow thread: https://stackoverflow.com/questions/2678551/when-to-encode-space-to-plus-or-20. This is surpisingly a widely misunderstood, and hotly contested topic. TL;DR + is used for application/x-www-form-urlencoded.

Comment (Atbhatt) [reply lib/marketo_api/api/base.rb]: Interesting, lots of debate on that StackOverflow thread. Good to know what he is referencing but worth starting a conversation with Steven to discuss design standardization on Marketo's API param usage, also guidelines that we can apply to our own public API. Thanks for following up on that @kuiro5.

**Comment by Atbhatt** [comment]
The version on master is `0.0.7-alpha`, any reason for the jump to `0.1.1-alpha`?

**Pull request merged**